packages/cron/test/cargo.spec.js

```diff
@@ -0,0 +1,163 @@
+/* eslint-env mocha */
+import assert from 'assert'
+import { createCargoDag, createUpload, createUser, createUserAuthKey, dbEndpoint, getUpload, listUploads, randomCid, token } from '@web3-storage/db/test-utils'
+import { DBClient } from '@web3-storage/db'
+import { updateDagSizes } from '../src/jobs/dagcargo.js'
+import { getPg } from '../src/lib/utils.js'
+
+const env = {
+  DEBUG: '*',
+  ENV: 'dev',
+  PG_CONNECTION: 'postgresql://postgres:postgres@localhost:5432/postgres',
+  RO_PG_CONNECTION: 'postgresql://postgres:postgres@localhost:5432/postgres',
+  DATABASE: 'postgres',
+  PG_REST_URL: 'http://localhost:3000',
+  PG_REST_JWT: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJzdXBhYmFzZSIsImlhdCI6MTYwMzk2ODgzNCwiZXhwIjoyNTUwNjUzNjM0LCJyb2xlIjoic2VydmljZV9yb2xlIn0.necIJaiP7X2T2QjGeV-FhpkizcNTX8HjDDBAxpgQTEI'
+}
+
+function getToBeUpdatedNull (cItem) {
+  return cItem[1] === null && // It's null in web3.storage
+    cItem[2] !== null// It's has a size in cargo
+}
+
+function getToBeUpdatedWrong (cItem) {
+  return cItem[1] !== cItem[2] && //  web3.storage value is different from cargo
+    cItem[2] !== null // It's has a size in cargo
+}
+
+describe('Fix dag sizes migration', () => {
+  /** @type {DBClient} */
+  let dbClient
+  let user
+  let authKey
+  let contentItems
+  let roPg
+  let rwPg
+
+  async function updateDagSizesWrp ({ user, after = new Date(1990, 1, 1), limit = null }) {
+    const allUploadsBefore = await listUploads(dbClient, user._id)
+    await updateDagSizes({
+      roPg,
+      rwPg,
+      after,
+      limit
+    })
+    const allUploadsAfter = await listUploads(dbClient, user._id)
+    const updatedCids = allUploadsAfter.filter((uAfter) => {
+      const beforeUpload = allUploadsBefore.find((uBefore) => uAfter.cid === uBefore.cid)
+      return beforeUpload?.dagSize !== uAfter.dagSize
+    }).map(u => u.cid)
+
+    return {
+      allUploadsAfter,
+      allUploadsBefore,
+      updatedCids
+    }
+  }
+
+  before(async () => {
+    dbClient = new DBClient({
+      endpoint: dbEndpoint,
+      token,
+      postgres: true
+    })
+    rwPg = getPg(env, 'rw')
+    roPg = getPg(env, 'ro')
+    await rwPg.connect()
+    await roPg.connect()
+  })
+
+  after(async () => {
+    await rwPg.end()
+    await roPg.end()
+  })
+
+  beforeEach(async () => {
+    user = await createUser(dbClient)
+    authKey = await createUserAuthKey(dbClient, user._id)
+    // cid, public.content.dag_size, cargo.dag.size_actual
+    contentItems = [
+      [await randomCid(), 120, 120],
+      [await randomCid(), 105, 1000], // To be updated
+      [await randomCid(), 110, 110],
+      [await randomCid(), 125, 90], // To be updated
+      [await randomCid(), 0, 90], // To be updated
+      [await randomCid(), 100, null],
+      [await randomCid(), null, null],
+      [await randomCid(), 0, 0]
```

Comment:
More code but these tests would be much more readable if these were objects like this for example:

```suggestion
      [{ cid: await randomCid(), dagSize: 0, dagSizeActual: 0 }]
```

...you could even `map` these arrays into that format if you wanted.

Reply:
Really good point, not sure why I ended using an array here.
Updated.